### PR TITLE
Add row selectors template samples for all grids

### LINKED ISSUE #1363: Add row numbering demos for all the grids
### Description  
Create demos for all the grids with row numbering as a row selector column for the grid.

Similar to [that](https://www.igniteui.com/help/iggrid-configuring-row-selectors#checkboxes-preview) look, just we need to clarify if we want to have both row number and checbox selector, or it is better and clear to template it only with row number. @StefanIvanov what do you think?

### LINKED ISSUE #1417: Add excel-like row numbering demo
### Description
Add excel-like row numbering demos for IgxGrid

### PR DESCRIPTION
**Pull request by wnvko**
Create a new sample showing how to use row selector template in order to add row index in the row selectors.

Create and a grid sample showing how to make the grid's row selection to look like Excel's row selection.

Additionally force some samples to use data from `grid/services`, instead of using local copy of the same data.

Closes #1363 
Closes #1417

### CODE REVIEW

File: src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.html [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+            </div>
+        </ng-template>
+
+        <ng-template igxRowSelector let-rowContext>
+            <div class="row-selector">
+                <span>{{ rowContext.index + 1 }}</span>
+            </div>
+        </ng-template>
+    </igx-grid>
+</div>
```
Comment (jackofdiamond5) [inline src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.html]:
```suggestion
</div>

```

File: src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                </div>
+                <igx-checkbox class="row-selector-checkbox"
+                    [checked]="rowContext.selected"
+                    disableRipple="true" [disableTransitions]="myGrid.disableTransitions"
+                    [aria-label]="myGrid.rowCheckboxAriaLabel">
+                </igx-checkbox>
+            </div>
+        </ng-template>
+    </igx-grid>
+</div>
```
Comment (jackofdiamond5) [inline src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.html]:
```suggestion
</div>

```

File: src/app/hierarchical-grid/hierarchical-grid-selection-template-numbers/hierarchical-grid-selection-template-numbers.component.html [hunk 3]
```diff
@@ -0,0 +1,46 @@
+<igx-hierarchical-grid #hierarchicalGrid class="hgrid" [data]="localData" [rowSelection]="'multiple'"
+    [cellSelection]="'none'" [autoGenerate]="false" [height]="'600px'" [width]="'100%'" [paging]="true">
+    <igx-column field="Artist"></igx-column>
+    <igx-column field="Debut"></igx-column>
+    <igx-column field="GrammyNominations" header="Grammy Nominations"></igx-column>
+    <igx-column field="GrammyAwards" header="Grammy Awards"></igx-column>
+
+    <igx-row-island [key]="'Albums'" [autoGenerate]="false" [rowSelectable]="true">
```
Comment (jackofdiamond5) [inline src/app/hierarchical-grid/hierarchical-grid-selection-template-numbers/hierarchical-grid-selection-template-numbers.component.html]:
```suggestion
    <igx-row-island [key]="'Albums'" [autoGenerate]="false" [rowSelection]="'multiple'">
```

File: src/app/hierarchical-grid/hierarchical-grid-selection-template-numbers/hierarchical-grid-selection-template-numbers.component.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        <igx-column field="USBillboard200" header="US Billboard 200"></igx-column>
+        <igx-row-island [key]="'Songs'" [autoGenerate]="false" [rowSelectable]="true">
+            <igx-column field="Number" header="No."></igx-column>
+            <igx-column field="Title"></igx-column>
+            <igx-column field="Released"></igx-column>
+            <igx-column field="Genre"></igx-column>
+        </igx-row-island>
+    </igx-row-island>
+
+    <igx-row-island [key]="'Tours'" [autoGenerate]="false" [rowSelectable]="true">
```
Comment (jackofdiamond5) [inline src/app/hierarchical-grid/hierarchical-grid-selection-template-numbers/hierarchical-grid-selection-template-numbers.component.html]:
```suggestion
    <igx-row-island [key]="'Tours'" [autoGenerate]="false" [rowSelection]="'multiple'">
```

File: src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.html [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+            </div>
+        </ng-template>
+
+        <ng-template igxRowSelector let-rowContext>
+            <div class="row-selector">
+                <div class="row-selector-index">
+                    <span>{{ rowContext.index + 1 }}</span>
+                </div>
+                <igx-checkbox class="row-selector-checkbox"
+                    [checked]="rowContext.selected"
```
Comment (jackofdiamond5) [inline src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.html]:
```suggestion
                    [checked]="rowContext.selected"
                    [readonly]="true"
```

File: src/app/hierarchical-grid/hierarchical-grid-selection-template-numbers/hierarchical-grid-selection-template-numbers.component.html [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        </div>
+    </ng-template>
+
+    <ng-template igxRowSelector let-rowContext>
+        <div class="row-selector">
+            <div class="row-selector-index">
+                <span>{{ rowContext.index + 1 }}</span>
+            </div>
+            <igx-checkbox class="row-selector-checkbox"
+                [checked]="rowContext.selected"
```
Comment (jackofdiamond5) [inline src/app/hierarchical-grid/hierarchical-grid-selection-template-numbers/hierarchical-grid-selection-template-numbers.component.html]:
```suggestion
                [checked]="rowContext.selected"
                [readonly]="true"
```

File: src/app/tree-grid/tree-grid-selection-template-numbers/tree-grid-selection-template-numbers.component.html [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+            </div>
+        </ng-template>
+
+        <ng-template igxRowSelector let-rowContext>
+            <div class="row-selector">
+                <div class="row-selector-index">
+                    <span>{{ rowContext.index + 1 }}</span>
+                </div>
+                <igx-checkbox class="row-selector-checkbox"
+                    [checked]="rowContext.selected"
```
Comment (jackofdiamond5) [inline src/app/tree-grid/tree-grid-selection-template-numbers/tree-grid-selection-template-numbers.component.html]:
```suggestion
                    [checked]="rowContext.selected"
                    [readonly]="true"
```

File: live-editing/configs/GridConfigGenerator.ts [hunk 8]
```diff
@@ -338,6 +350,36 @@ export class GridConfigGenerator implements IConfigGenerator {
             })
         }));
 
+        configs.push(new Config({
+            component: GridSelectionTemplateExcelComponent,
+            additionalFiles: ["/src/app/grid/services/financialData.ts"],
```
Comment (damyanpetev) [inline live-editing/configs/GridConfigGenerator.ts]: Samples uses static data though..  (`import { DATA } from "../../data/customers";`), this shouldn't be needed.

File: live-editing/configs/GridConfigGenerator.ts [hunk 9]
```diff
@@ -338,6 +350,36 @@ export class GridConfigGenerator implements IConfigGenerator {
...
+            appModuleConfig: new AppModuleConfig({
+                imports: [HttpClientModule, IgxAvatarModule, IgxBadgeModule, IgxButtonModule,
```
Comment (damyanpetev) [inline live-editing/configs/GridConfigGenerator.ts]: Umm, any specific reason the Excel style sample that uses just the grid (header template is not even an icon) would need avatar, http, badge and so on.

File: live-editing/configs/GridConfigGenerator.ts [hunk 10]
```diff
@@ -338,6 +350,36 @@ export class GridConfigGenerator implements IConfigGenerator {
...
+                ngImports: [IgxAvatarModule, IgxBadgeModule, IgxButtonModule, IgxGridModule,
+                    IgxIconModule, IgxInputGroupModule, IgxProgressBarModule,
+                    IgxRippleModule, IgxSwitchModule, HttpClientModule],
+                ngProviders: []
+            })
+        }));
+
+        configs.push(new Config({
+            component: GridSelectionTemplateNumbersComponent,
+            additionalFiles: ["/src/app/grid/services/financialData.ts"],
```
Comment (damyanpetev) [inline live-editing/configs/GridConfigGenerator.ts]: Same as above

File: live-editing/configs/GridConfigGenerator.ts [hunk 11]
```diff
@@ -338,6 +350,36 @@ export class GridConfigGenerator implements IConfigGenerator {
...
+            appModuleConfig: new AppModuleConfig({
+                imports: [HttpClientModule, IgxAvatarModule, IgxBadgeModule, IgxButtonModule,
```
Comment (damyanpetev) [inline live-editing/configs/GridConfigGenerator.ts]: Same, I see just grid and checkbox

File: src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.html [hunk 12]
```diff
@@ -0,0 +1,25 @@
+<div class="grid__wrapper">
+    <igx-grid #myGrid [data]="data" [primaryKey]="'ID'" [rowSelection]="'multiple'" [cellSelection]="'none'"
```
Comment (damyanpetev) [inline src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.html]:
```suggestion
    <igx-grid #myGrid [data]="data" primaryKey="ID" rowSelection="multiple" cellSelection="none"
```

File: src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.html [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+        [paging]="true" [height]="'530px'" width="100%">
```
Comment (damyanpetev) [inline src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.html]: And so on

File: src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.html [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+        <igx-column [field]="'ContactName'" [width]="'20%'"></igx-column>
```
Comment (damyanpetev) [inline src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.html]: Applies for everything:
```suggestion
        <igx-column field="ContactName" width="20%"></igx-column>
```

File: src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.html [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+        <igx-column [field]="'Country'" [width]="'20%'"></igx-column>
+        <igx-column [field]="'City'" [width]="'20%'" dataType="number"></igx-column>
+        <igx-column [field]="'PostalCode'" [width]="'20%'" dataType="number"></igx-column>
+        <igx-column [field]="'CompanyName'" [width]="'20%'" dataType="number"></igx-column>
+
+        <ng-template igxHeadSelector let-headContext>
+            <div class="header-selector">
+                <div *ngIf="headContext.selectedCount === headContext.totalCount && headContext.totalCount !== 0"
```
Comment (damyanpetev) [inline src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.html]: You can just use one div and [`[ngClass]`](https://angular.io/guide/template-syntax#ngclass) to swap the classes.

File: src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.scss [hunk 16]
```diff
@@ -0,0 +1,56 @@
+@import "~igniteui-angular/lib/core/styles/themes/index";
+@include igx-core();
```
Comment (damyanpetev) [inline src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.scss]:
```suggestion
```

File: src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.scss [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+@include igx-theme($default-palette);
```
Comment (damyanpetev) [inline src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.scss]: No need to re-emit the default theme :)
```suggestion
```

File: src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.html (hunk 15 above)
Comment (damyanpetev) [reply src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.html]: Also you might be able to do this with one of the material icons: https://material.io/resources/icons/?icon=signal_cellular_4_bar&style=baseline

File: src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.scss [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+        @include triangle($default-arrow-hover-color);
+    }
+}
+
+.row-selector {
+    @include selector(center, center);
+}
+
+::ng-deep {
+    .igx-grid__cbx-selection {
```
Comment (damyanpetev) [inline src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.scss]: This tells me there's something wrong with the way the template is setup, this shouldn't be required from the users.

File: src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.scss [hunk 19]
```diff
@@ -0,0 +1,43 @@
...
+
+.header-image {
+    width: 40px;
+    height: 40px;
+}
+
+.row-selector {
+    width: 70px;
+    display: flex;
+    justify-content: center;
```
Comment (damyanpetev) [inline src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.scss]: Yeah, those items don't really fit inside a box 50 box with internal padding where the number is 40. The checkbox is way off:
image
You can remove the extra div (put the index span and checkbox as siblings) and go:
```suggestion
    justify-content: space-evenly;
```
With removed padding it looks centered and quite even on all pages.

File: src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.scss [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+    align-items: center;
+    padding: 0 10px;
```
Comment (damyanpetev) [inline src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.scss]: per above
```suggestion
```

File: src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.html [hunk 21]
```diff
@@ -0,0 +1,30 @@
...
+                    [readonly]="true"
+                    disableRipple="true" [disableTransitions]="myGrid.disableTransitions"
+                    [aria-label]="myGrid.rowCheckboxAriaLabel">
```
Comment (damyanpetev) [inline src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.html]: Not sure what that is, but it's not a public prop:
image

File: src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.scss [hunk 22]
```diff
@@ -0,0 +1,43 @@
...
+    height: 100%;
+}
+
+.row-selector-checkbox {
+    pointer-events: none;
```
Comment (damyanpetev) [inline src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.scss]: Not sure what that is for.. does `disableRipple` on the checkbox not work?

File: src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.scss [hunk 23]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+.row-selector-index {
```
Comment (damyanpetev) [inline src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.scss]: You can re-purpose this for the span index and it'll need **only** a `cursor: default;` so it doesn't do text selection on the index.

**Comment by damyanpetev** [comment]
Logged issue for the default padding: https://github.com/IgniteUI/igniteui-angular/issues/5764

File: src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.scss (hunk 18 above)
Comment (wnvko) [reply src/app/grid/grid-sample-selection-template-excel/grid-sample-selection-template-excel.component.scss]: we should remove this once it is fixed in the component default template

File: src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.scss (hunk 22 above)
Comment (wnvko) [reply src/app/grid/grid-sample-selection-template-numbers/grid-sample-selection-template-numbers.component.scss]: leftover used before we made checkbox readonly :)

**Pull request merged**